x-pack/plugins/enterprise_search/public/applications/enterprise_search/components/trial_callout/trial_callout.tsx

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+
+import { useValues } from 'kea';
+import moment from 'moment';
+
+import { EuiCallOut, EuiLink, EuiSpacer } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n/react';
+
+import { LicensingLogic } from '../../../shared/licensing';
+
+export const TrialCallout: React.FC = () => {
+  const { license, isTrial } = useValues(LicensingLogic);
+
+  if (!isTrial) return null;
+
+  const title = (
+    <FormattedMessage
+      id="xpack.enterpriseSearch.trialCalloutTitle"
+      defaultMessage="Your Elastic Stack Trial license, which enables Platinum features, expires in {days, plural, one {# day} other {# days}}. {link}"
+      values={{
+        days: moment(license?.expiryDateInMillis).diff(moment({ hours: 0 }), 'days'),
+        link: (
+          <EuiLink href="https://www.elastic.co/subscriptions" target="_blank">
+            <strong>Learn more about Elastic Stack licenses.</strong>
+          </EuiLink>
+        ),
+      }}
+    />
+  );
```

Comment:
Awesome work here with the pluralization and license expiring logic! i18n request: I think I'd actually suggest keeping the `{link}` as a totally separate string/clause, since it's its own full sentence. Also partially because nested strings can be tricky - for instance the actual `Learn more about Elastic Stack licenses` text itself is missing i18n localization here.

```suggestion
  const title = (
    <>
      <FormattedMessage
        id="xpack.enterpriseSearch.trialCalloutTitle"
        defaultMessage="Your Elastic Stack Trial license, which enables Platinum features, expires in {days, plural, one {# day} other {# days}}."
        values={{
          days: moment(license?.expiryDateInMillis).diff(moment({ hours: 0 }), 'days'),
        }}
      />{' '}
      <EuiLink href="https://www.elastic.co/subscriptions" target="_blank">
        <strong>
          <FormattedMessage
            id="xpack.enterpriseSearch.trialCalloutLink"
            defaultMessage="Learn more about Elastic Stack licenses."
          />
        </strong>
      </EuiLink>
    </>
  );
```

Reply:
One other quick q - is the `<strong>` tag intentional vs the `<u>` underline tag that ent-search appears to be using? No worries if so, just wanted to check the difference in styles

Reply:
Wow. TIL. I was looking at the original CSS when I did this. Didn't even check the markup:

![image](https://user-images.githubusercontent.com/1869731/117052288-2ac16780-acdd-11eb-8309-a351ff57f685.png)

Reply:
https://github.com/elastic/kibana/pull/99122/commits/27c9938972d922f007be664f3abdf572d6eb63be

Reply:
Looks great!